packages/fxa-auth-server/test/local/routes/password.js

```diff
@@ -71,6 +76,152 @@ describe('/password', () => {
     mocks.unMockAccountEventsManager();
   });
 
+  describe('/forgot/send_otp', () => {
+    const mockConfig = {
+      passwordForgotOtp: {
+        enabled: true,
+        digits: 8,
+        ttl: 300,
+      },
+    };
+    const mockCustoms = mocks.mockCustoms();
+    const uid = uuid.v4({}, Buffer.alloc(16)).toString('hex');
+    const mockDB = mocks.mockDB({
+      email: TEST_EMAIL,
+      uid,
+    });
+    const mockMailer = mocks.mockMailer();
+    const mockMetricsContext = mocks.mockMetricsContext();
+    const mockLog = mocks.mockLog('ERROR', 'test', {
+      stdout: {
+        on: sinon.spy(),
+        write: sinon.spy(),
+      },
+      stderr: {
+        on: sinon.spy(),
+        write: sinon.spy(),
+      },
+    });
+    mockLog.flowEvent = sinon.spy(() => {
+      return Promise.resolve();
+    });
+    const mockRedis = {
+      set: sinon.stub(),
+      get: sinon.stub(),
+      del: sinon.stub(),
+    };
+    const mockStatsd = { increment: sinon.stub() };
+
+    it('sends an OTP when enabled', () => {
+      const passwordRoutes = makeRoutes({
+        config: mockConfig,
+        customs: mockCustoms,
+        db: mockDB,
+        mailer: mockMailer,
+        metricsContext: mockMetricsContext,
+        log: mockLog,
+        authServerCacheRedis: mockRedis,
+        statsd: mockStatsd,
+      });
+
+      const mockRequest = mocks.mockRequest({
+        log: mockLog,
+        payload: {
+          email: TEST_EMAIL,
+          metricsContext: {
+            deviceId: 'wibble',
+            flowId:
+              'F1031DF1031DF1031DF1031DF1031DF1031DF1031DF1031DF1031DF1031DF103',
+            flowBeginTime: Date.now() - 1,
+          },
+        },
+        query: {},
+        metricsContext: mockMetricsContext,
+      });
+      return runRoute(
+        passwordRoutes,
+        '/password/forgot/send_otp',
+        mockRequest
+      ).then((response) => {
+        assert.equal(
+          mockDB.accountRecord.callCount,
+          1,
+          'db.accountRecord was called once'
+        );
+        sinon.assert.calledOnce(mockRedis.set);
+
+        // an eight digit code was set
+        // TODO FXA-7852 check that the same code was pass to the email
+        assert.match(mockRedis.set.args[0][1], /\d{8}/);
```

Comment:
Q: Does this need a `^` and `$` anchor? Otherwise it might match if the code was 8, or 9, or 10+ digits?

```js
const str = 1234567890;
const re = /\d{8}/;
console.log(re.exec(str)); // `["12345678"]`
```

Reply:
Oh yes, good catch.